sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/DiagnosticExtensions.cs

```diff
@@ -12,5 +15,79 @@ public static string GetAsciiString(this ArraySegment<byte> arraySegment)
         {
             return arraySegment.Array == null ? string.Empty : Encoding.ASCII.GetString(arraySegment.Array, arraySegment.Offset, arraySegment.Count);
         }
+
+        public static void SetMessageTags(this DiagnosticScope scope, IEnumerable<ServiceBusReceivedMessage> messages)
```

Comment:
This makes me want an abstract base class for messages..

Reply:
A `Message`? 😃

Reply:
More likely `ServiceBusMessageBase`. The question is whether users will commonly need to do stuff like this. If not, we can work around this by using an internal interface.

Reply:
Personally, I'd keep a hard segmentation between outgoing and incoming messages as the default position and let feedback guide any decision to use a common public base.   

My rationale is that messaging is typically used to facilitate communication between systems or the individual components of a distributed system.  The role of producer and consumer is most often an exclusive one.  You may see a system consume messages from others and then publish its own set of messages in response - but those would likely be considered as being in a different domain and unlikely to be something where you wanted to treat them equally and perform a common operation en masse.

In my experience it would be a very rare case for a system to both publish messages and consume the messages that it published.  For that scenario, I can see where grouping messages regardless of incoming/outgoing may be occasionally helpful, but view that as very niche and wouldn't optimize for it.

Reply:
> In my experience it would be a very rare case for a system to both publish messages and consume the messages that it published. For that scenario, I can see where grouping messages regardless of incoming/outgoing may be occasionally helpful, but view that as very niche and wouldn't optimize for it.

Granted I don't know do you mean by a 'system', my customers do have internal events published publish and consumed by the same system. They usually do that from different endpoints/processes. And internal endpoint pub/sub is not as rare as it might be thought of. Just adding the observations from the wild world out there 🙂

Reply:
That's a fair call-out.  😄 

I'm being loose and sloppy with terminology here.  My intent was to use `system` as an isolated unit, either stand-alone or as one part of a larger distributed system.  You wouldn't typically send a message and read it back within the same area of responsibility/isolation since you're not crossing a boundary.  While I've seen some examples of using localized queues/topics in unique cases, such as a fan-out/fan-in strategy to manage work scheduling, I'd call them niche.  

Agreed in that decoupling different discrete parts of the same application/ecosystem (bounded contexts) and having them communicate through message passing is something that I'd see as a core scenario.   I'm including something like having a set of endpoints participating in a workflow using a message broker to transition states in this bucket of my mental model, and considering each step of the workflow to be a discrete unit of isolation.

Reply:
I think the plan of record is to leave out the base for now, and add it in if needed in the future.
In terms of cleaning up this code, I can try to use an internal interface.